### PR TITLE
Coach

### PR DESCRIPTION
**Pull request by Montana-Pfeifer**
Added winningest and worst coach to stat tracker merged with main

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -108,4 +108,26 @@ def lowest_total_score
...
+    coach_games = Hash.new { |hash, key| hash[key] = { wins: 0, games: 0}}
+    @all_game_teams.each do |game_team|
+      coach = game_team.head_coach
+      coach_games[coach][:games] += 1
+      coach_games[coach][:wins] += 1 if game_team.result == "WIN"
+    end
+      coach_games.transform_values do |stats| 
+        games = stats[:games]
+        games > 0 ? ((stats[:wins].to_f / games) * 100).round : 0
+    end
```
Comment (Crosswolfv1) [inline lib/stat_tracker.rb]: I think there is some alignment issues here for indents 118-121

**Review by Crosswolfv1 — CHANGES_REQUESTED** [review]
Changes look good, small indentation issues check notes please

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -108,4 +108,26 @@ def lowest_total_score
...
+  end
+
+  def winningest_coach
+    coach_win_percentages = coach_win_percentages()
```
Comment (wally-yawn) [inline lib/stat_tracker.rb]: Don't need the () here

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -108,4 +108,26 @@ def lowest_total_score
...
+    coach_win_percentages.max_by { |coach, win_percentage| win_percentage}.first
+  end
+
+  def worst_coach
+    coach_win_percentages = coach_win_percentages()
```
Comment (wally-yawn) [inline lib/stat_tracker.rb]: Don't need the () here

**Review by wally-yawn — APPROVED** [review]

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -108,4 +108,26 @@ def lowest_total_score
...
+      coach_games[coach][:wins] += 1 if game_team.result == "WIN"
+    end
+      coach_games.transform_values do |stats| 
+        games = stats[:games]
+        games > 0 ? ((stats[:wins].to_f / games) * 100).round : 0
+      end
+  end
+
+  def winningest_coach
+    coach_win_percentages = coach_win_percentages
```
Comment (ejdesh) [inline lib/stat_tracker.rb]: What is the purpose of this line?

Comment (ejdesh) [reply lib/stat_tracker.rb]: it looks like an error because it is trying to assign to a method.

File: lib/stat_tracker.rb [hunk 5]
```diff
@@ -108,4 +108,26 @@ def lowest_total_score
...
+    coach_win_percentages.max_by { |coach, win_percentage| win_percentage}.first
+  end
+
+  def worst_coach
+    coach_win_percentages = coach_win_percentages()
+    coach_win_percentages.min_by { |coach, win_percentage| win_percentage}.first
```
Comment (ejdesh) [inline lib/stat_tracker.rb:129]: in the event of a tie, you're going to return the first coach. Is this intended?

File: lib/stat_tracker.rb (hunk 1 above)
Comment (Montana-Pfeifer) [reply lib/stat_tracker.rb]: I fixed alignment

File: lib/stat_tracker.rb (hunk 2 above)
Comment (Montana-Pfeifer) [reply lib/stat_tracker.rb]: In Ruby, if you want to call a method that has no arguments, you need to include parentheses when invoking it, especially if you are assigning the return value to a variable with the same name as the method.

File: lib/stat_tracker.rb (hunk 3 above)
Comment (Montana-Pfeifer) [reply lib/stat_tracker.rb]: In Ruby, if you want to call a method that has no arguments, you need to include parentheses when invoking it, especially if you are assigning the return value to a variable with the same name as the method.

File: lib/stat_tracker.rb (hunk 4 above)
Comment (Montana-Pfeifer) [reply lib/stat_tracker.rb]: In Ruby, if you want to call a method that has no arguments, you need to include parentheses when invoking it, especially if you are assigning the return value to a variable with the same name as the method.

File: lib/stat_tracker.rb (hunk 5 above)
Comment (Montana-Pfeifer) [reply lib/stat_tracker.rb:129]: yes because it will show the min coach first

File: lib/stat_tracker.rb (hunk 4 above)
Comment (ejdesh) [reply lib/stat_tracker.rb]: Are you sure @Montana-Pfeifer ?
image

**Review by Crosswolfv1 — APPROVED** [review]

**Pull request merged**